plasmapy/mathematics/tests/test_dispersion.py

```diff
@@ -1,92 +1,167 @@
+    zeros = np.array([
+            1.991_466_842_833_879_6 - 1.354_810_128_112_006_2j,
+            2.691_149_024_251_438_8 - 2.177_044_906_089_615_9j,
+            3.235_330_868_352_816_5 - 2.784_387_613_230_428_2j,
+            3.697_309_702_468_468_4 - 3.287_410_789_389_848_6j,
+            4.106_107_284_682_632_1 - 3.725_948_719_445_790_4j],
+            dtype=np.complex128)
 
-    with pytest.raises(TypeError):
-        plasma_dispersion_func('')
+    for zero in zeros:
+        Z = plasma_dispersion_func(zero)
+        assert np.isclose(Z, 0 + 0j, atol=1e-15 * (1 + 1j)), \
+            ("A zero of the plasma dispersion function is expected at ζ = "
+             f"{zero}, but plasma_dispersion_func({zero}) is equal to {Z}.")
 
-    with pytest.raises(u.UnitsError):
-        plasma_dispersion_func(6*u.m)
 
-    with pytest.raises(ValueError):
-        plasma_dispersion_func(np.inf)
+# ζ, expected
+plasma_disp_deriv_table = [
+    (0, -2),
+    (1, 0.152_318 - 1.304_10j),
+    (1j, -0.484_257),
+    (1.2 + 4.4j, -0.397_561e-1 - 0.217_392e-1j),
+    (9j, plasma_dispersion_func_deriv(9j * u.dimensionless_unscaled)),
+    (5.4 - 3.1j, 0.012_449_1 + 0.023_138_3j),
+    (9.9 - 10j, 476.153 + 553.121j),
+    (5 + 7j, -4.591_20e-3 - 0.012_610_4j),
+    (4.5 - 10j, 0.260_153e37 - 0.211_814e37j)]
 
 
-def test_plasma_dispersion_func_deriv():
-    r"""Test the implementation of plasma_dispersion_func_deriv against
-    tabulated results from Fried & Conte (1961)."""
+@pytest.mark.parametrize('ζ, expected', plasma_disp_deriv_table)
+def test_plasma_dispersion_func_deriv(ζ, expected):
+    r"""Test the implementation of plasma_dispersion_func_deriv
+    against tabulated results and an analytical relationship from
+    Fried & Conte (1961)."""
 
-    atol = 2e-6*(1 + 1j)
+    Z_deriv = plasma_dispersion_func_deriv(ζ)
 
-    assert np.isclose(plasma_dispersion_func_deriv(0), -2, atol=atol, rtol=0),\
-        "Z'(0) not consistent with tabulated values"
+    assert np.isclose(Z_deriv, expected, atol=5e-5*(1+1j), rtol=5e-6), \
+        (f"The derivative of the plasma dispersion function does not match "
+         f"the expected value for ζ = {ζ}.  The value of "
+         f"plasma_dispersion_func_deriv({ζ}) equals {Z_deriv} whereas the "
+         f"expected value is {expected}.  The difference between the actual "
+         f"and expected results is {Z_deriv - expected}.")
 
-    assert np.isclose(plasma_dispersion_func_deriv(1), 0.152318 - 1.30410j,
-                      atol=atol, rtol=0), \
-        "Z'(1) not consistent with tabulated values"
+    Z = plasma_dispersion_func(ζ)
+    Z_deriv_characterization = -2 * (1 + ζ*Z)
 
-    assert np.isclose(plasma_dispersion_func_deriv(1j),
-                      -0.484257, atol=atol, rtol=0), \
-        "Z'(1j) not consistent with tabulated values"
+    assert np.isclose(Z_deriv, Z_deriv_characterization, rtol=1e-15), \
+        (f"The relationship that Z'(ζ) = -2 * [1 + ζ * Z(ζ)] is not "
```

Comment:
I am wondering if sympy would be useful here. As mentioned in #176

Reply:
How would you see it used here?

Reply:
To nicely display that equation without symbol voodoo.